src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -437,23 +425,29 @@ struct consume_to_store {
     }
 
     void operator()(model::record record) {
-        auto key = schema_key_from_iobuf(record.release_key());
+        auto key = from_json_iobuf<schema_key_handler<>>(record.release_key());
         if (key.keytype == topic_key_type::schema) {
-            vassert(key.magic == 1, "Schema key magic is unknown");
-            auto val = schema_value_from_iobuf(record.release_value());
-            vlog(
-              plog.debug,
-              "Inserting key: subject: {}, version: {} ",
-              key.sub(),
-              key.version);
-            auto res = _store.insert(val.sub, val.schema, val.type);
-            vassert(res.id == val.id, "Schema_id mismatch");
-            vassert(res.version == val.version, "Schema_version mismatch");
+            apply(
+              std::move(key),
+              from_json_iobuf<schema_value_handler<>>(record.release_value()));
         } else {
             vlog(
               plog.warn, "Ignoring keytype: {}", to_string_view(key.keytype));
         }
     }
+
+    void apply(schema_key key, schema_value val) {
+        vassert(key.magic == 1, "Schema key magic is unknown");
```

Comment:
assert for a runtime var? why not `result<>` ?  seems like a public method tho. or do you want a private method here? seems like a very easy way to crash redpanda.